packages/smart-contracts/packages/staking/contracts/Staking.sol

```diff
@@ -707,23 +707,25 @@ contract Staking is
         uint256[] memory bandIds = s_stakerBands[user];
         uint256 bandsAmount = bandIds.length;
 
-        // Loop through all bands that user owns and delete data
-        for (uint256 bandIndex; bandIndex < bandsAmount; bandIndex++) {
-            // Effects: delete all band data
-            delete s_bands[bandIds[bandIndex]];
-        }
+        if (bandIds.length != 0) {

```

Comment:
I think it makes more sense to revert with custom error if no bands exist.

Reply:
Then many of the test cases will fail, due to us wanting to remove he vesting user from vesting, but they have not staked. So we can not remove them at all

Reply:
Then I suggest to move event emit inside if block